README.md
>> `cd react-code` 
>> `npm install`
>> `cd client`
>> `npm install yarn --global`
>> `yarn install`
>> `npm start`

client/CHANGELOG.md
# Change Log

## [2.0.1] 2022-07-04

- Update dependencies
- Migration to React 18
- Migration to sass from node-sass

## [2.0.0] 2021-01-21

### IMPORTANT

- We have updated this product from Bootstrap 3 to Bootstrap 4, so in essence, this is a new product
- For this, we have followed the guidelines from [here](https://react-bootstrap.github.io/migrating/) and [here](https://getbootstrap.com/docs/4.0/migration/)
- We also did not add Bootstrap variables as part of our styles (we will do so in one of our next updates, probably in version 3.0.0 when we'll add Bootstrap 5)

### Bug fixing

- Renamed all files to `.js`, instead of `.jsx`
- Refactor all files from class components to functional ones, so now, you can use React Hooks
- https://github.com/creativetimofficial/light-bootstrap-dashboard-react/issues/62
- https://github.com/creativetimofficial/light-bootstrap-dashboard-react/issues/58
- https://github.com/creativetimofficial/light-bootstrap-dashboard-react/issues/52
- https://github.com/creativetimofficial/light-bootstrap-dashboard-react/issues/50

### Major style changes

- Since the update from Bootstrap 3 to Bootstrap 4, all styles have been changed

### Deleted components

- src/components/Card/Card.jsx (we'll use the simple React Bootstrap tags instead)
- src/components/CustomButton/CustomButton.jsx (we'll use the simple React Bootstrap tags instead)
- src/components/CustomCheckbox/CustomCheckbox.jsx (we'll use the simple React Bootstrap tags instead)
- src/components/CustomRadio/CustomRadio.jsx (we'll use the simple React Bootstrap tags instead)
- src/components/FormInputs/FormInputs.jsx (we'll use the simple React Bootstrap tags instead)
- src/components/StatsCard/StatsCard.jsx (we'll use the simple React Bootstrap tags instead)
- src/components/Tasks/Tasks.jsx (we'll use the simple React Bootstrap tags instead)
- src/components/UserCard/UserCard.jsx (we'll use the simple React Bootstrap tags instead)

### Added components

### Deleted dependencies

- react-notification-system (we'll use react-notification-alert from now on)
- @types/googlemaps
- @types/markerclustererplus
- @types/react
- react-google-maps (we'll use simple Google Maps API with Vanilla JS)
- react-toggle (we'll use the simple React-Bootstrap Form.Check as a switch instead)

### Added dependencies

- react-notification-alert@0.0.13 (instead of the react-notification-system)
- reactstrap@8.9.0 (as part of the react-notification-alert)
- jquery@3.5.1 (since Bootstrap requires it)
- @fortawesome/fontawesome-free@5.15.2 (For easier update of the package - instead of a link inside public/index.html)
- gulp-append-prepend@1.0.8 (For Creative Tim licenses)
- gulp@4.0.2 (For Creative Tim licenses)

### Updated dependencies

```
bootstrap           3.3.7   →    4.6.0
node-sass          4.12.0   →   4.14.1
react              16.8.6   →   17.0.1
react-bootstrap    0.32.4   →    1.4.3
react-chartist     0.13.3   →   0.14.3
react-dom          16.8.6   →   17.0.1
react-router        5.0.0   →    5.2.0
react-router-dom    5.0.0   →    5.2.0
react-scripts       3.0.0   →    4.0.1
typescript          3.4.3   →    4.1.3
```

### Warning

**The TypeScript and jQuery dependencies are installed only to stop console warnings on install. They are not actually used in our product. So the product is not based on TypeScript or jQuery!**
_The following warnings will appear when running the installation command, but they do not affect the UI or the functionality of the product (they will be solved in our next update):_

```
npm WARN react-chartist@0.14.3 requires a peer of react@^0.14.9 || ^15.0.0 || ^16.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN react-popper@1.3.7 requires a peer of react@0.14.x || ^15.0.0 || ^16.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN create-react-context@0.3.0 requires a peer of react@^0.14.0 || ^15.0.0 || ^16.0.0 but none is installed. You must install peer dependencies yourself.
```

_If they will persist in our 2.4.\* version, we will drop their usages and replace them with other plugins._
_In development mode, some of the above plugins will throw a warning because they still use React v16 syntax. If the error will persist in our 2.4.\* version, we will drop their usage and replace them with other plugins._

### Upgrade details

- Updated all packages from `package.json` using `npm-upgrade`, link here: https://www.npmjs.com/package/npm-upgrade
- Renamed all `ControlLabel` components to `FormLabel` (this refers to the react-bootstrap components)
- Renamed all `Grid` components to `Container` (this refers to the react-bootstrap components)
- Rename all `HelpBlock` components to `FormText`
- Rename all `MenuItem` components to `Dropdown.Item` and `NavDropdown.Item`
- Rename all Panel components to Card
- Deleted variables folder
- Change the usage of `React Big Calendar` to the new API, so instead of

```
import BigCalendar from "react-big-calendar";
```

- And

```
const localizer = BigCalendar.momentLocalizer(moment);
```

- We will have:

```
import { Calendar as BigCalendar, momentLocalizer } from "react-big-calendar";
```

- And

```
const localizer = momentLocalizer(moment);
```

- Change Panels with new react-bootstrap components
- For Col componets, replaced `size={number1} sizeOffest={number2}` with `size={{span: number1, offset: number2}}` where size can be `xs`, `sm`, `md`, `lg`
- Changed `bsStyle` to `variant`
- Changed `bsSize` to `size`
- Replace `<Navbar.Header>` with `<div className=“navbar-header”>`
- Replace `<Navbar.Form>` with `<div className="navbar-search-form navbar-form navbar-left">`
- Replace `<FormGroup>` with `<Form.Group>`
- Replace `<InputGroup.Addon>` with `<InputGroup.Prepend><InputGroup.Text>` and `<InputGroup.Append><InputGroup.Text>`

client/src/index.js
/*!

=========================================================
* Light Bootstrap Dashboard React - v2.0.1
=========================================================

* Product Page: https://www.creative-tim.com/product/light-bootstrap-dashboard-react
* Copyright 2022 Creative Tim (https://www.creative-tim.com)
* Licensed under MIT (https://github.com/creativetimofficial/light-bootstrap-dashboard-react/blob/master/LICENSE.md)

* Coded by Creative Tim

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

*/
import React from "react";
import ReactDOM from "react-dom/client";

import { BrowserRouter, Route, Switch, Redirect } from "react-router-dom";

import "bootstrap/dist/css/bootstrap.min.css";
import "./assets/css/animate.min.css";
import "./assets/scss/light-bootstrap-dashboard-react.scss?v=2.0.0";
import "./assets/css/demo.css";
import "@fortawesome/fontawesome-free/css/all.min.css";
import ContextProvider from "./components/context/ContextProvider";
import AdminLayout from "layouts/Admin.js";

const root = ReactDOM.createRoot(document.getElementById("root"));

root.render(
  <ContextProvider>
  <BrowserRouter>
    <Switch>
      <Route path="/home" render={(props) => <AdminLayout {...props} />} />
      <Redirect from="/" to="/home/users" />
    </Switch>
  </BrowserRouter>
  </ContextProvider>
);


client/src/routes.js
/*!

=========================================================
* Light Bootstrap Dashboard React - v2.0.1
=========================================================

* Product Page: https://www.creative-tim.com/product/light-bootstrap-dashboard-react
* Copyright 2022 Creative Tim (https://www.creative-tim.com)
* Licensed under MIT (https://github.com/creativetimofficial/light-bootstrap-dashboard-react/blob/master/LICENSE.md)

* Coded by Creative Tim

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

*/
// import Dashboard from "views/Dashboard.js";
import Home from "views/Home.js";
import Detail from "views/Detail.js";
import Register from "views/Register";
// import TableList from "views/TableList.js";
// import Notifications from "views/Notifications.js";

const homeRoutes = [
  {
    path: "/users",
    name: "Home",
    icon: "nc-icon nc-circle-09",
    component: Home,
    layout: "/home",
    sitebar: true
  },
  {
    path: "/user/detail/:id",
    name: "Detail",
    icon: null,
    component: Detail,
    layout: "/home",
    sitebar: false
  },
  {
    path: "/user/register",
    name: "Register",
    icon: null,
    component: Register,
    layout: "/home",
    sitebar: false
  }
];

export default homeRoutes;


client/src/views/Home.js
import React, { useState, useEffect, useContext } from 'react'
import { NavLink } from 'react-router-dom';
import { adddata, deldata } from '../components/context/ContextProvider';
import { updatedata } from '../components/context/ContextProvider'
import {
    Container,
    Row,
    Col
  } from "react-bootstrap";
const Home = () => {
    const [getuserdata, setUserdata] = useState([]);
    const { udata, setUdata } = useContext(adddata);
    const {updata, setUPdata} = useContext(updatedata);
    const {dltdata, setDLTdata} = useContext(deldata);
    const getdata = async () => {
        const res = await fetch("https://crudappreactjs.herokuapp.com/getdata", {
            method: "GET",
            headers: {
                "Content-Type": "application/json"
            }
        });
        const data = await res.json();
        if (res.status === 422 || !data) {
            console.log("error ");
        } else {
            setUserdata(data)
            console.log("get data");
        }
    }

    useEffect(() => {
        getdata();
    }, [])
    const deleteuser = async (id) => {
        const res2 = await fetch(`https://crudappreactjs.herokuapp.com/deleteuser/${id}`, {
            method: "DELETE",
            headers: {
                "Content-Type": "application/json"
            }
        });
        const deletedata = await res2.json();
        console.log(deletedata);
        if (res2.status === 422 || !deletedata) {
            console.log("error");
        } else {
            console.log("user deleted");
            setDLTdata(deletedata)
            getdata();
        }
    }
    return (
        <>
        {
            udata ?
                <>
                    <div class="alert alert-success alert-dismissible fade show" role="alert">
                        <strong>{udata.name}</strong>  added succesfully!
                        <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
                    </div>
                </> : ""
        }
        {
            updata ?
                <>
                    <div class="alert alert-success alert-dismissible fade show" role="alert">
                        <strong>{updata.name}</strong>  updated succesfully!
                        <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
                    </div>
                </> : ""
        }
        {
            dltdata ?
                <>
                    <div class="alert alert-danger alert-dismissible fade show" role="alert">
                        <strong>{dltdata.name}</strong>  deleted succesfully!
                        <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
                    </div>
                </> : ""
        }
        <Container>
            <Row>
                <Col>
                    <div className="add_btn mt-2 mb-2">
                        <NavLink to="user/register" className="btn btn-sm">Add data</NavLink>
                    </div>
                    <table class="table" striped hover>
                        <thead>
                            <tr>
                                <th scope="col">id</th>
                                <th scope="col">Username</th>
                                <th scope="col">email</th>
                                <th scope="col">Job</th>
                                <th scope="col">Number</th>
                                <th scope="col"></th>
                            </tr>
                        </thead>
                        <tbody>
                            {
                                getuserdata.map((element, id) => {
                                    return (
                                        <>
                                            <tr>
                                                <th scope="row">{id + 1}</th>
                                                <td>{element.name}</td>
                                                <td>{element.email}</td>
                                                <td>{element.work}</td>
                                                <td>{element.mobile}</td>
                                                <td className="d-flex justify-content-between">
                                                    <NavLink to={`user/detail/${element._id}`}><button className="btn btn-sm">view</button></NavLink>
                                                    <NavLink to={`user/detail/${element._id}`}><button className="btn btn-sm">edit</button></NavLink>
                                                    <NavLink to={`/`} onClick={() => deleteuser(element._id)}>
                                                        <button className="btn-danger btn-sm text-white" size="sm">delete</button>
                                                    </NavLink>
                                                </td>
                                            </tr>
                                        </>
                                    )
                                })
                            }
                        </tbody>
                    </table>
                </Col>
            </Row>
        </Container>
        </>
    )
}

export default Home

client/src/layouts/Admin.js
/*!

=========================================================
* Light Bootstrap Dashboard React - v2.0.1
=========================================================

* Product Page: https://www.creative-tim.com/product/light-bootstrap-dashboard-react
* Copyright 2022 Creative Tim (https://www.creative-tim.com)
* Licensed under MIT (https://github.com/creativetimofficial/light-bootstrap-dashboard-react/blob/master/LICENSE.md)

* Coded by Creative Tim

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

*/
import React, { Component } from "react";
import { useLocation, Route, Switch } from "react-router-dom";

import AdminNavbar from "components/Navbars/AdminNavbar";
import Footer from "components/Footer/Footer";
import Sidebar from "components/Sidebar/Sidebar";
import routes from "routes.js";

function Admin() {
  const [color, setColor] = React.useState("black");
  const [hasImage, setHasImage] = React.useState(true);
  const location = useLocation();
  const mainPanel = React.useRef(null);
  const getRoutes = (routes) => {
    return routes.map((prop, key) => {
      if (prop.layout === "/home") {
        return (
          <Route
            path={prop.layout + prop.path}
            render={(props) => <prop.component {...props} />}
            key={key}
          />
        );
      } else {
        return null;
      }
    });
  };
  React.useEffect(() => {
    document.documentElement.scrollTop = 0;
    document.scrollingElement.scrollTop = 0;
    mainPanel.current.scrollTop = 0;
    if (
      window.innerWidth < 993 &&
      document.documentElement.className.indexOf("nav-open") !== -1
    ) {
      document.documentElement.classList.toggle("nav-open");
      var element = document.getElementById("bodyClick");
      element.parentNode.removeChild(element);
    }
  }, [location]);
  return (
    <>
      <div className="wrapper">
        <Sidebar color={"black"} routes={routes} />
        <div className="main-panel" ref={mainPanel}>
          <AdminNavbar />
          <div className="content">
            <Switch>{getRoutes(routes)}</Switch>
          </div>
          <Footer />
        </div>
      </div>
    </>
  );
}

export default Admin;


client/src/views/Register.js
import React, { useContext, useState } from 'react'
import { useHistory } from 'react-router-dom'
import { adddata } from '../components/context/ContextProvider';
import {
    Button,
    Card,
    Form,
    Container,
    Row,
    Col
  } from "react-bootstrap";
const Register = () => {
    const { udata, setUdata } = useContext(adddata);
    const history = useHistory();
    const [inpval, setINP] = useState({
        name: "",
        email: "",
        age: "",
        mobile: "",
        work: "",
        add: "",
        desc: ""
    })
    const setdata = (e) => {
        console.log(e.target.value);
        const { name, value } = e.target;
        setINP((preval) => {
            return {
                ...preval,
                [name]: value
            }
        })
    }
    const addinpdata = async (e) => {
        e.preventDefault();
        const { name, email, work, add, mobile, desc, age } = inpval;
        const res = await fetch("https://crudappreactjs.herokuapp.com/register", {
            method: "POST",
            headers: {
                "Content-Type": "application/json"
            },
            body: JSON.stringify({
                name, email, work, add, mobile, desc, age
            })
        });
        const data = await res.json();
        console.log(data);
        if (res.status === 422 || !data) {
            console.log("error ");
            alert("error");
        } else {
            history.push("/")
            setUdata(data)
            console.log("data added");
        }
    }
    return (
        <>
        <Container fluid>
          <Row>
            <Col md="8">
              <Card>
                <Card.Header>
                  <Card.Title as="h4">Welcome Harsh Pathak!</Card.Title>
                </Card.Header>
                <Card.Body>
                  <Form>
                    <Row>
                      <Col className="pr-1" md="5">
                        <Form.Group>
                          <label>Company</label>
                          <Form.Control
                            name="work"
                            defaultValue={inpval.work}
                            placeholder="Company"
                            type="text"
                            onChange={setdata}
                          ></Form.Control>
                        </Form.Group>
                      </Col>
                      <Col className="px-1" md="3">
                        <Form.Group>
                          <label>Name</label>
                          <Form.Control
                            name="name"
                            defaultValue={inpval.name}
                            placeholder="Username"
                            type="text"
                            onChange={setdata}
                          ></Form.Control>
                        </Form.Group>
                      </Col>
                      <Col className="pl-1" md="4">
                        <Form.Group>
                          <label htmlFor="exampleInputEmail1">
                            Email address
                          </label>
                          <Form.Control
                            name="email"
                            defaultValue={inpval.email}
                            placeholder="Email"
                            type="email"
                            onChange={setdata}
                          ></Form.Control>
                        </Form.Group>
                      </Col>
                    </Row>
                    <Row>
                      <Col md="12">
                        <Form.Group>
                          <label>Address</label>
                          <Form.Control
                            name="add"
                            defaultValue={inpval.add}
                            placeholder="Home Address"
                            type="text"
                            onChange={setdata}
                            ></Form.Control>
                        </Form.Group>
                      </Col>
                    </Row>
                    <Row>
                      <Col className="pr-1" md="4">
                        <Form.Group>
                          <label>Age</label>
                          <Form.Control
                            name="age"
                            defaultValue={inpval.age}
                            placeholder="Age"
                            type="number"
                            onChange={setdata}
                            ></Form.Control>
                        </Form.Group>
                      </Col>
                      <Col className="pl-1" md="4">
                        <Form.Group>
                          <label>Mobile</label>
                          <Form.Control
                            name="mobile"
                            defaultValue={inpval.mobile}
                            placeholder="+91 123456"
                            type="text"
                            onChange={setdata}
                            ></Form.Control>
                        </Form.Group>
                      </Col>
                    </Row>
                    <Row>
                      <Col md="12">
                        <Form.Group>
                          <label>Description</label>
                          <Form.Control
                            name="desc"
                            cols="80"
                            defaultValue={inpval.desc}
                            placeholder="Here can be your description"
                            rows="4"
                            as="textarea"
                            onChange={setdata}
                            ></Form.Control>
                        </Form.Group>
                      </Col>
                    </Row>
                    <Row className="mt-2">
                        <Col>
                            <Button
                                className="btn-fill pull-right"
                                type="submit"
                                variant="info"
                                onClick={addinpdata}
                                >
                                Submit
                            </Button>
                        </Col>
                    </Row>
                    <div className="clearfix"></div>
                  </Form>
                </Card.Body>
              </Card>
            </Col>
          </Row>
        </Container>
        </>
    )
}
export default Register;


client/src/components/Sidebar/Sidebar.js
/*!

=========================================================
* Light Bootstrap Dashboard React - v2.0.1
=========================================================

* Product Page: https://www.creative-tim.com/product/light-bootstrap-dashboard-react
* Copyright 2022 Creative Tim (https://www.creative-tim.com)
* Licensed under MIT (https://github.com/creativetimofficial/light-bootstrap-dashboard-react/blob/master/LICENSE.md)

* Coded by Creative Tim

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

*/
import React, { Component } from "react";
import { useLocation, NavLink } from "react-router-dom";

import { Nav } from "react-bootstrap";

import logo from "assets/img/reactlogo.png";

function Sidebar({ color, image, routes }) {
  const location = useLocation();
  const activeRoute = (routeName) => {
    return location.pathname.indexOf(routeName) > -1 ? "active" : "";
  };
  return (
    <div className="sidebar" data-color={color}>
      <div className="sidebar-wrapper">
        <div className="logo d-flex align-items-center justify-content-start">
          <a>
            Manage Users
          </a>
        </div>
        <Nav>
          {routes.map((prop, key) => {
            if (!prop.redirect && prop.sitebar)
              return (
                <li
                  className={
                    prop.upgrade
                      ? "active active-pro"
                      : activeRoute(prop.layout + prop.path)
                  }
                  key={key}
                >
                  <NavLink
                    to={prop.layout + prop.path}
                    className="nav-link"
                    activeClassName="active"
                  >
                    <i className={prop.icon} />
                    <p>{prop.name}</p>
                  </NavLink>
                </li>
              );
            return null;
          })}
        </Nav>
      </div>
    </div>
  );
}

export default Sidebar;


client/src/views/Detail.js
import React, { useContext, useEffect, useState } from 'react'
import { useParams,useHistory } from 'react-router-dom'
import { updatedata } from '../components/context/ContextProvider'
import {
    Button,
    Card,
    Form,
    Container,
    Row,
    Col
  } from "react-bootstrap";

const Detail = () => {
    const {setUPdata} = useContext(updatedata)
    const history = useHistory("");
    const [inpval, setINP] = useState({
        name: "",
        email: "",
        age: "",
        mobile: "",
        work: "",
        add: "",
        desc: ""
    })
    const setdata = (e) => {
        console.log(e.target.value);
        const { name, value } = e.target;
        setINP((preval) => {
            return {
                ...preval,
                [name]: value
            }
        })
    }
    const { id } = useParams("");
    const getdata = async () => {
        const res = await fetch(`https://crudappreactjs.herokuapp.com/getuser/${id}`, {
            method: "GET",
            headers: {
                "Content-Type": "application/json"
            }
        });
        const data = await res.json();
        console.log(data);
        if (res.status === 422 || !data) {
            console.log("error");
        } else {
            setINP(data)
            console.log("get data");
        }
    }
    useEffect(() => {
        getdata();
    }, []);
    const updateuser = async(e)=>{
        e.preventDefault();
        const {name,email,work,add,mobile,desc,age} = inpval;
        const res2 = await fetch(`https://crudappreactjs.herokuapp.com/updateuser/${id}`,{
            method: "PATCH",
            headers: {
                "Content-Type": "application/json"
            },
            body:JSON.stringify({
                name,email,work,add,mobile,desc,age
            })
        });
        const data2 = await res2.json();
        // console.log(data2);
        if(res2.status === 422 || !data2){
            alert("fill the data");
        }else{
            history.push("/")
            setUPdata(data2);
        }
    }
    return (
        <>
        <Container fluid>
          <Row>
            <Col md="8">
              <Card>
                <Card.Header>
                  <Card.Title as="h4">Welcome Harsh Pathak!</Card.Title>
                </Card.Header>
                <Card.Body>
                  <Form>
                    <Row>
                      <Col className="pr-1" md="5">
                        <Form.Group>
                          <label>Company (disabled)</label>
                          <Form.Control
                            name="work"
                            defaultValue={inpval.work}
                            disabled
                            placeholder="Company"
                            type="text"
                          ></Form.Control>
                        </Form.Group>
                      </Col>
                      <Col className="px-1" md="3">
                        <Form.Group>
                          <label>Name</label>
                          <Form.Control
                            name="name"
                            defaultValue={inpval.name}
                            placeholder="Username"
                            type="text"
                            onChange={setdata}
                          ></Form.Control>
                        </Form.Group>
                      </Col>
                      <Col className="pl-1" md="4">
                        <Form.Group>
                          <label htmlFor="exampleInputEmail1">
                            Email address
                          </label>
                          <Form.Control
                            name="email"
                            defaultValue={inpval.email}
                            placeholder="Email"
                            type="email"
                            onChange={setdata}
                          ></Form.Control>
                        </Form.Group>
                      </Col>
                    </Row>
                    <Row>
                      <Col md="12">
                        <Form.Group>
                          <label>Address</label>
                          <Form.Control
                            name="add"
                            defaultValue={inpval.add}
                            placeholder="Home Address"
                            type="text"
                            onChange={setdata}
                            ></Form.Control>
                        </Form.Group>
                      </Col>
                    </Row>
                    <Row>
                      <Col className="pr-1" md="4">
                        <Form.Group>
                          <label>Age</label>
                          <Form.Control
                            name="age"
                            defaultValue={inpval.age}
                            placeholder="Age"
                            type="number"
                            onChange={setdata}
                            ></Form.Control>
                        </Form.Group>
                      </Col>
                      <Col className="pl-1" md="4">
                        <Form.Group>
                          <label>Mobile</label>
                          <Form.Control
                            name="mobile"
                            defaultValue={inpval.mobile}
                            placeholder="+91 123456"
                            type="text"
                            onChange={setdata}
                            ></Form.Control>
                        </Form.Group>
                      </Col>
                    </Row>
                    <Row>
                      <Col md="12">
                        <Form.Group>
                          <label>Description</label>
                          <Form.Control
                            name="desc"
                            cols="80"
                            defaultValue={inpval.desc}
                            placeholder="Here can be your description"
                            rows="4"
                            as="textarea"
                            onChange={setdata}
                            ></Form.Control>
                        </Form.Group>
                      </Col>
                    </Row>
                    <Row className="mt-2">
                        <Col>
                            <Button
                                className="btn-fill pull-right"
                                type="submit"
                                variant="info"
                                onClick={updateuser}
                                >
                                Update Profile
                            </Button>
                        </Col>
                        <Col>
                            <Button
                                className="btn-danger pull-right"
                                type="submit"
                                variant="info"
                                onClick={() => deleteuser(inpval._id)}
                                >
                                Delete Profile
                            </Button>
                        </Col>
                    </Row>
                        
                    <div className="clearfix"></div>
                  </Form>
                </Card.Body>
              </Card>
            </Col>
            <Col md="4">
              <Card className="card-user">
                <div className="card-image">
                  <img
                    alt="..."
                    src={require("assets/img/photo-1431578500526-4d9613015464.jpeg")}
                  ></img>
                </div>
                <Card.Body>
                  <div className="author">
                    <a href="#pablo" onClick={(e) => e.preventDefault()}>
                      <img
                        alt="..."
                        className="avatar border-gray"
                        src={require("assets/img/faces/face-3.jpg")}
                      ></img>
                      <h5 className="title">{inpval.name}</h5>
                    </a>
                    <p className="description">{'+91 ' + inpval.mobile}</p>
                  </div>
                  <p className="description text-center">
                    {inpval.desc}
                  </p>
                </Card.Body>
                <hr></hr>
                <div className="button-container mr-auto ml-auto">
                  <Button
                    className="btn-simple btn-icon"
                    href="#pablo"
                    onClick={(e) => e.preventDefault()}
                    variant="link"
                  >
                    <i className="fab fa-facebook-square"></i>
                  </Button>
                  <Button
                    className="btn-simple btn-icon"
                    href="#pablo"
                    onClick={(e) => e.preventDefault()}
                    variant="link"
                  >
                    <i className="fab fa-twitter"></i>
                  </Button>
                  <Button
                    className="btn-simple btn-icon"
                    href="#pablo"
                    onClick={(e) => e.preventDefault()}
                    variant="link"
                  >
                    <i className="fab fa-google-plus-square"></i>
                  </Button>
                </div>
              </Card>
            </Col>
          </Row>
        </Container>
      </>
    )
}

export default Detail